File: Go/hubuco_email_verifier.go
package main

import (
    "os"
    "fmt"
    "net/http"
    "io/ioutil"
    "encoding/json"
)

const
    // !!! PUT YOUR API KEY HERE !!!
    api_key = "YOUR_API_KEY"


type RespType struct {
    ResultCode int `json:"resultcode"`
    Error string `json:"error"`
}


func main() {
    if len(os.Args) != 2 {
    fmt.Printf("usage: %s <email>\n", os.Args[0])
    os.Exit(1)
    }

    email := os.Args[1]

    response, err := http.Get("https://api.hubuco.com/api/v3/?api="+api_key+"&"+"email="+email)
    if err != nil {
        panic(err)
    }
    defer response.Body.Close()
    contents, err := ioutil.ReadAll(response.Body)
    if err != nil {
        panic(err)
    }

    res := RespType{}
    if err := json.Unmarshal([]byte(contents), &res); err != nil {
        panic(err)
    }

    switch res.ResultCode {
    case 1:
        fmt.Println("Ok")
    case 2:
        fmt.Println("Catch All")
    case 3:
        fmt.Println("Unknown")
    case 4:
        fmt.Println("Error: " + res.Error)
    case 5:
        fmt.Println("Disposable")
    case 6:
        fmt.Println("Invlaid")
    }

}

File: Python3/hubuco_email_verifier.py
import sys
import requests


# !!!PUT YOUR API KEY HERE!!!
api_key="YOUR_API_KEY"


if len(sys.argv) != 2:
    print("usage: ", sys.argv[0], " <email>")
    sys.exit(1)

email = sys.argv[1]

j = requests.get("https://api.hubuco.com/api/v3/?api="+api_key+"&"+"email="+email).json()

c = j['resultcode']

if c == 1:
    print("Ok")
elif c == 2:
    print("Catch All")
elif c == 3:
    print("Unknown")
elif c == 4:
    print("Error: " + j['error'])
elif c == 5:
    print("Disposable")
elif c == 6:
    print("Invlaid")

